sdks/java/io/elasticsearch/src/main/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIO.java

```diff
@@ -0,0 +1,744 @@
+ * {@code withPassword()}.
+ *
+ * <p>You can also specify a query on the {@code read()} using {@code withQuery()}.
+ *
+ * <h3>Writing to Elasticsearch</h3>
+ *
+ * <p>ElasticsearchIO supports sink to write documents (as JSON String).
+ *
+ * <p>To configure Elasticsearch sink, similar to the read, you have to provide a connection
+ * configuration. For instance:
+ *
+ * <pre>{@code
+ *
+ *  pipeline
+ *    .apply(...)
+ *    .apply(ElasticsearchIO.write().withConnectionConfiguration(
+ *       ElasticsearchIO.ConnectionConfiguration.create("http://host:9200", "my-index", "my-type")
+ *    )
+ *
+ * }</pre>
+ *
+ * <p>Optionally, you can provide {@code withBatchSize()} and {@code withBatchSizeMegaBytes()}
+ * to specify the size of the write batch in number of documents or in megabytes.
+ */
+public class ElasticsearchIO {
+
+  public static Read read() {
+    return new AutoValue_ElasticsearchIO_Read.Builder().setScrollKeepalive("5m").build();
+  }
+
+  public static Write write() {
+    return new AutoValue_ElasticsearchIO_Write.Builder()
+        .setMaxBatchSize(1000L)
+        .setMaxBatchSizeMegaBytes(5)
+        .build();
+  }
+
+  private ElasticsearchIO() {}
+
+  private static JsonObject parseResponse(Response response) throws IOException {
+    InputStream content = response.getEntity().getContent();
+    InputStreamReader inputStreamReader = new InputStreamReader(content, "UTF-8");
+    JsonObject jsonObject = new Gson().fromJson(inputStreamReader, JsonObject.class);
+    return jsonObject;
+  }
+
+  /** A POJO describing a connection configuration to Elasticsearch. */
+  @AutoValue
+  public abstract static class ConnectionConfiguration implements Serializable {
+
+    abstract List<String> getAddresses();
+
+    @Nullable
+    abstract String getUsername();
+
+    @Nullable
+    abstract String getPassword();
+
+    abstract String getIndex();
+
+    abstract String getType();
+
+    abstract Builder builder();
+
+    @AutoValue.Builder
+    abstract static class Builder {
+      abstract Builder setAddresses(List<String> addresses);
+
+      abstract Builder setUsername(String username);
+
+      abstract Builder setPassword(String password);
+
+      abstract Builder setIndex(String index);
+
+      abstract Builder setType(String type);
+
+      abstract ConnectionConfiguration build();
+    }
+
+    /**
+     * Create Elasticsearch connection configuration.
+     *
+     * @param addresses list of addresses of Elasticsearch nodes
+     * @param index the index toward which the requests will be issued
+     * @param type the document type toward which the requests will be issued
+     * @return the connection configuration object
+     */
+    public static ConnectionConfiguration create(String[] addresses, String index, String type) {
+      checkArgument(
+          addresses != null,
+          "ConnectionConfiguration.create(addresses, index, type) " + "called with null address");
```

Comment:
Weird, I now use google autoformat. Previous autoformatter has split the strings and google auto format has put them in same line without merging them. I'll fix them all and check for google auto-format conf. Thanks for noting